db/kvstorewithbuffer.go

```diff
@@ -168,6 +168,10 @@ func (kvb *kvStoreWithBuffer) MustDelete(ns string, key []byte) {
 	kvb.buffer.Delete(ns, key, "failed to delete %x in %s", key, ns)
 }
 
+func (kvb *kvStoreWithBuffer) Filter(ns string, c Condition) ([][]byte, [][]byte, error) {
+	return kvb.store.Filter(ns, c)
```

Comment:
need to filter the entities in buffer as well

Reply:
ok